include/BBTrack.h

```diff
@@ -108,13 +111,21 @@ protected slots:
 
 protected:
 	void paintEvent( QPaintEvent * );
+	void resizeEvent( QResizeEvent * re )
```

Comment:
>Also I don't think the declaration of resizeEvent should be virtual void in SampleTCOView if it contains code.

https://github.com/LMMS/lmms/blob/de7d83d15897407b6300e2e3475ce6f7c62903a2/include/Pattern.h#L186

Its like that in Pattern.cpp as well. I'm not so good at c++, so I don't really get the difference between `virtual void` and `void`.

>I see m_needsUpdate is declared in each class. That might be redundant. See this: http://stackoverflow.com/a/14270779/3196753

I don't quite understand. I should define a private member variable `m_needsUpdate` in `TrackContentObjectView`, and then make functions that allow the child classes to access that variable?